### PR TITLE
DOC-1010: Add clarification note of CRDB antipatterns

### PR DESCRIPTION
**Pull request by nickvigilante**
Fixes DOC-1010

### CODE REVIEW

**Review by hand-crdb — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/14451)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @nickvigilante)

___
*[`v22.1/migrate-from-postgres.md` line 24 at r1](https://reviewable.io/reviews/cockroachdb/docs/14451#-N6FGghZ6UPB5N_b3RX8:-N6FGghZ6UPB5N_b3RX9:b-yv2ro3) ([raw file](https://github.com/cockroachdb/docs/blob/16194defd4abfaa739a15df66c0e278ebf56cfc7/v22.1/migrate-from-postgres.md#L24)):*
> ```Markdown
> PostgreSQL and CockroachDB have differences in best practices surrounding [primary keys](primary-key.html) on tables. While it's common to see sequences and auto-incrementing primary keys in PostgreSQL, these features can cause hotspots within your cluster when reading or writing large amounts of data. The preferred approaches for  are using [multi-column primary keys](performance-best-practices-overview.html#use-multi-column-primary-keys) or using the [`UUID`](uuid.html) datatype for primary key columns.
> 
> If you are working with a table that must be indexed on sequential keys, use [hash-sharded indexes](hash-sharded-indexes.html).
> ```

Overall, this new section is great.  Regarding this particular sentence...

Whether or not to use hash-sharded indexes is more complex than that.  Consider rewording to something like:

If you are working with a table that must be indexed on sequential keys, consider using [hash-sharded indexes](hash-sharded-indexes.html).  As part of this, it is a good idea to do performance testing with and without hash-sharded indexes to see which works best for your application.

**Review by hand-crdb — COMMENTED** [review]
> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/14451)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @nickvigilante)

File: v22.1/migrate-from-postgres.md [hunk 1]
```diff
@@ -15,6 +15,16 @@ The examples pull real data from [Amazon S3](https://aws.amazon.com/s3/). They u
...
 
+## Pre-migration considerations
+
+### Primary keys
+
+PostgreSQL and CockroachDB have differences in best practices surrounding [primary keys](primary-key.html) on tables. While it's common to see sequences and auto-incrementing primary keys in PostgreSQL, these features can cause hotspots within your cluster when reading or writing large amounts of data. The preferred approaches for  are using [multi-column primary keys](performance-best-practices-overview.html#use-multi-column-primary-keys) or using the [`UUID`](uuid.html) datatype for primary key columns.
+
+If you are working with a table that must be indexed on sequential keys, consider using [hash-sharded indexes](hash-sharded-indexes.html). As part of this, we recommend doing thorough performance testing with and without hash-sharded indexes to see which works best for your application.
+
+For further information, see [Unique ID best practices](performance-best-practices-overview.html#unique-id-best-practices) and [3 Basic Rules for Choosing Indexes](https://www.cockroachlabs.com/blog/how-to-choose-db-index-keys/)/
```
Comment (stbof) [inline v22.1/migrate-from-postgres.md:26]:
```suggestion
For further information, see [Unique ID best practices](performance-best-practices-overview.html#unique-id-best-practices) and [3 Basic Rules for Choosing Indexes](https://www.cockroachlabs.com/blog/how-to-choose-db-index-keys/).
```

File: v22.1/migrate-from-postgres.md [hunk 2]
```diff
@@ -15,6 +15,16 @@ The examples pull real data from [Amazon S3](https://aws.amazon.com/s3/). They u
 To migrate from PostgreSQL to CockroachDB using the AWS Database Migration Service, see [Migrate with AWS Database Migration Service (DMS)](aws-dms.html).
 {{site.data.alerts.end}}
 
+## Pre-migration considerations
+
+### Primary keys
+
+PostgreSQL and CockroachDB have differences in best practices surrounding [primary keys](primary-key.html) on tables. While it's common to see sequences and auto-incrementing primary keys in PostgreSQL, these features can cause hotspots within your cluster when reading or writing large amounts of data. The preferred approaches for  are using [multi-column primary keys](performance-best-practices-overview.html#use-multi-column-primary-keys) or using the [`UUID`](uuid.html) datatype for primary key columns.
```
Comment (stbof) [inline v22.1/migrate-from-postgres.md:22]:
```suggestion
PostgreSQL and CockroachDB have differences in best practices surrounding [primary keys](primary-key.html) on tables. While it's common to see sequences and auto-incrementing primary keys in PostgreSQL, these features can cause hotspots within your cluster when reading or writing large amounts of data. Cockroach Labs recommends that you use [multi-column primary keys](performance-best-practices-overview.html#use-multi-column-primary-keys) or the [`UUID`](uuid.html) datatype for primary key columns.
```

**Review by stbof — CHANGES_REQUESTED** [review]
See suggestions.

**Review by stbof — APPROVED** [review]

**Pull request merged**